src/main.rs

```diff
@@ -48,6 +53,7 @@ fn select_emoji() -> Option<&'static str> {
             Key::Char('\n') => break,
             Key::Up | Key::Char('k') | Key::Char('K') => selected = selected.prev_variant().unwrap_or(CommitType::last_variant()),
             Key::Down | Key::Char('j') | Key::Char('J') => selected = selected.next_variant().unwrap_or(CommitType::first_variant()),
+            Key::Char(key @ '1' ... '5') => { selected = get_emoji_at_index(key.to_string().parse::<u8>().unwrap() - 1); },
```

Comment:
I think you should be able to do `key - '1' as u8` instead of casting to string and parsing it back to a number. This works because a char is just an integer representing a character, and since the numerals are consecutively ordered (which is why `'1' ... '5'` works) you can just subtract the value of the first one to get a 0-based index.

Together with my other comments up there I would also extend the range to `key @ '1' ... '9'` and do something like:

```rust
Key::Char(key @ '1' ... '9') => CommitType::at_index(key - '1' as u8).and_then(|commit_type| selected = commit_type)
```

to only set `selected` when an actual commit type was found. This way you don't have to adjust the `'5'` when the number of commit types grows or shrinks...